client/spec.yaml

```diff
@@ -0,0 +1,286 @@
+openapi: 3.0.0
+info:
+  description: The Observatorium API
+  version: "0.0.1"
+  title: Observatorium API
+  license:
+    name: Apache 2.0
+    url: "http://www.apache.org/licenses/LICENSE-2.0.html"
+tags:
+  - name: observatoriumv1
+    description: Calls related to Observatorium
+# Keeping all the API paths here, but likely need to separate it out to another file,
+# once paths for other signals are also added.
+paths:
+  /api/metrics/v1/{tenant}/api/v1/rules/raw:
```

Comment:
I do not understand the meaning of the word "raw" here.

Do we have two formats for rules?  Under which circumstance would a client desire raw vs non-raw?

Reply:
We refer to the YAML format of the Prometheus Rules as "raw", as this is, in a sense, unprocessed. 

When these Rules are configured on a component like Thanos Ruler, they are "processed", i.e, some health data like "lastEvaluatedTime" or "lastError" is added to it which we can retrieve in JSON format via API call. Also, in the case of alerting Rules, we can get the alerts fired from the Rule and there are also some templated labels that are parsed! 

The circumstances under which this distinction is important for a client is when using the Rules API.
The client can configure their YAML Rules via the /rules/raw endpoint by sending a PUT request to Observatorium API and can see what is configured by sending a GET request to the same endpoint.

They can then retrieve health data + alerts or "processed" Rules, that they configured, in JSON format by sending a GET request to the /rules endpoint (Observatorium API just proxies request to Thanos Querier in this case, which gets this information from Thanos Ruler). 

Hope that makes it a bit clear! 🙂